crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -81,6 +81,18 @@ pub(crate) struct ReflectEnum<'a> {
     variants: Vec<EnumVariant<'a>>,
 }
 
+impl<'a> ReflectEnum<'a> {
+    /// Get a collection of types which are exposed to the reflection API
+    pub fn active_types(&self) -> Vec<syn::Type> {
+        Vec::default()
+    }
+
+    /// Get a collection of types which are ignored by the reflection API
+    pub fn ignored_types(&self) -> Vec<syn::Type> {
+        Vec::default()
+    }
+}
```

Comment:
I think that if we want this to work directly on `ReflectEnum`, we need it to work on `EnumVariant`. So we could do something like:

```rust
const EMPTY_FIELDS: &'static [StructField<'static>] = &[];

impl<'a> ReflectEnum<'a> {
    pub fn active_fields(&self) -> impl Iterator<Item = &StructField<'a>> {
        self.variants()
            .iter()
            .flat_map(|variant| variant.active_fields())
    }

    pub fn active_types(&self) -> Vec<syn::Type> {
        self.active_fields().map(|field| field.data.ty.clone())
            .collect()
    }
}

impl<'a> EnumVariant<'a> {
    pub fn active_fields(&self) -> impl Iterator<Item = &StructField<'a>> {
        match &self.fields {
            EnumVariantFields::Named(fields) | EnumVariantFields::Unnamed(fields) => fields.iter(),
            EnumVariantFields::Unit => EMPTY_FIELDS.iter(),
        }
    }

    pub fn active_types(&self) -> Vec<syn::Type> {
        self.active_fields()
            .map(|field| field.data.ty.clone())
            .collect()
    }
}
```

And of course, we'd need to include similar methods for the ignored fields.

Not sure if this actually works or not so we'll need to add some tests to check that.

Reply:
I'd prefer working on Enums in a different PR, if you don't mind?

Mainly because `#[reflect(ignore)]` isn't supported for enums right now; so i'd like to have a separate PR that both adds
`#[reflect(ignore)` for enums and enables reflection with generics.

Reply:
Actually, I guess we can just make it work for active fields, and ignore ignored_fields for now; and then work on ignored fields in a future PR!

Reply:
I just checked and it looks like `#[reflect(ignore)]` does work on enum fields? Could you verify this? I think we may have gotten it wrong on discord

Reply:
Also I was looking at #6767 and realized we actually don't need `EMPTY_FIELDS` if we just return `&[StructField]` directly instead of an iterator

Reply:
Really? I tested it locally and the `#[reflect(ignore)]` fields were still in the reflect impl

Reply:
Got enums working :)